recommender/engine/rl_agent/preprocessing/state_embedder.py

```diff
@@ -1,45 +1,136 @@
-from typing import Tuple
+# pylint: disable=missing-module-docstring, no-member, too-few-public-methods, no-name-in-module
+
+"""Implementation of the State Embedder"""
+
+from typing import Tuple, List
 
 import torch
+from torch import FloatTensor
+
+from recommender.engine.rl_agent.preprocessing.searchphrase_embedder import (
+    SearchPhraseEmbedder,
+)
+from recommender.engine.pre_agent.models import (
+    load_last_module,
+    USERS_AUTOENCODER,
+    SERVICES_AUTOENCODER,
+)
+from recommender.engine.rl_agent.preprocessing.filters_embedder import FiltersEmbedder
+from recommender.models import State, Service
+
+
+def _search_data_to_dict(search_data):
+    """
+    Transforms SearchData MongoEngine object into python dict.
+
+    Args:
+        search_data: SearchData Mongo Engine object.
 
-from models import State
+    Returns:
+        Python dict representation of the search data.
+
+    """
+    return dict(search_data.to_mongo())
 
 
 class StateEmbedder:
-    def __init__(self, user_embedder=None, service_embedder=None, search_phrase_embedder=None, filters_embedder=None):
-        self.user_embedder = user_embedder
-        self.service_embedder = service_embedder
-        self.search_phrase_embedder = search_phrase_embedder
-        self.filters_embedder = filters_embedder
+    """State Embedder"""
+
+    def __init__(
+        self,
+        user_embedder=None,
+        service_embedder=None,
+        search_phrase_embedder=None,
+        filters_embedder=None,
+    ):
+        if user_embedder is not None:
+            self.user_embedder = user_embedder
+        else:
+            self.user_embedder = load_last_module(USERS_AUTOENCODER).encoder
+
+        if service_embedder is not None:
+            self.service_embedder = service_embedder
+        else:
+            self.service_embedder = load_last_module(SERVICES_AUTOENCODER).encoder
 
-        # TODO: implement the rest of initialization (lazy embedders loading)
-        pass
+        if search_phrase_embedder is not None:
+            self.search_phrase_embedder = search_phrase_embedder
+        else:
+            self.search_phrase_embedder = SearchPhraseEmbedder()
+
+        if filters_embedder is not None:
```

Comment:
I guess you can use this kind of synatx:
`self.search_phrase_embedder = search_phrase_embedder if search_phrase_embedder is not None else  SearchPhraseEmbedder()` if you like.

Reply:
Here and in other places where module loading can be optional

Reply:
or even better: `self.search_phrase_embedder = search_phrase_embedder or SearchPhraseEmbedder()`